play/json-flattened-view-gen/src/main.rs

```diff
@@ -0,0 +1,110 @@
+use serde_json::value::Value;
+use structopt::StructOpt;
+
+#[derive(Debug)]
+enum PostgresType {
+    Text,
+    Other,
+}
+
+#[derive(thiserror::Error, Debug)]
+enum Error {
+    #[error("Sample json was not an serde_json::Value::Object.")]
+    NotAnObject,
+}
+
+fn flatten_object(map: Value) -> Result<Vec<(String, PostgresType)>, Error> {
+    if let Value::Object(m) = map {
+        let mut flattened = Vec::new();
+        for (key, val) in m.into_iter() {
+            flattened.extend(flatten_json(key, val));
+        }
+        Ok(flattened)
+    } else {
+        Err(Error::NotAnObject)
+    }
+}
+
+fn flatten_json(key: String, value: Value) -> Vec<(String, PostgresType)> {
+    let mut flattened: Vec<(String, PostgresType)> = Vec::new();
+    match value {
+        Value::Array(v) => {
+            v.into_iter().enumerate().for_each(|(i, v)| {
+                let new_key = format!("{}__{}", key, i);
+                flattened.extend(flatten_json(new_key, v));
+            });
+        }
+        Value::Bool(_) => flattened.push((key, PostgresType::Other)),
+        Value::Number(_) => flattened.push((key, PostgresType::Other)),
+        Value::String(_) => flattened.push((key, PostgresType::Text)),
+        Value::Object(v) => {
+            v.into_iter().for_each(|(i, v)| {
```

Comment:
`i` represents the key in the JSON object, rather than an index, right? Should be called something other than `i` in that case (perhaps `k` ?)

Reply:
Changed.